helix-term/src/commands.rs

```diff
@@ -1645,9 +1645,123 @@ fn search_selection(cx: &mut Context) {
     cx.editor.set_status(msg);
 }
 
+/// search with `matcher` in a background thread, returns a `JoinHandle` to be awaited on
+fn global_search_task(
+    file_picker_config: FilePickerConfig,
+    matcher: RegexMatcher,
+) -> JoinHandle<Vec<(usize, PathBuf)>> {
+    let searcher = SearcherBuilder::new()
+        .binary_detection(BinaryDetection::quit(b'\x00'))
+        .build();
+    let search_root =
+        std::env::current_dir().expect("Global search error: Failed to get current dir");
+    tokio::task::spawn_blocking(move || -> Vec<(usize, PathBuf)> {
```

Comment:
I wonder if a single tokio thread is good enough, ripgrep IIRC uses multiple threads and here we only use one. But I do think this is a good start, just wondering if we should directly switch to do multiple threads.

Reply:
`build_parallel` ensures that we do use multiple threads to search, we just wait for these threads to complete in a tokio 
 thread so that our main loop won't be blocked here.